codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/customize/ServerCodegenDecorator.kt

```diff
@@ -31,14 +33,20 @@ interface ServerCodegenDecorator : CoreCodegenDecorator<ServerCodegenContext, Se
      * constrained but the `ValidationException` shape is not attached to the operation's errors.
      */
     fun postprocessValidationExceptionNotAttachedErrorMessage(validationResult: ValidationResult) = validationResult
+
+    /**
+     * Injection point to allow a decorator to point to structures not in the service closure
```

Comment:
```suggestion
     * For each operation in the service's closure, this hook allows decorators to return a collection of structure shapes that will additionally be generated.
     * If a structure shape is in the service's closure, note that returning it here will cause for it to be generated more than once, making the resulting crate not compile (since it will contain more than one struct with the same name). Therefore, ensure that all the structure shapes returned by this method are not in the service's closure.
```